lib/more_core_extensions/core_ext/array/tableize.rb

```diff
@@ -118,7 +119,7 @@ def format_row(row, widths, justifications)
       end
 
       def format_field(field, width, justification)
-        field = field.to_s.gsub(/\n|\r/, '').slice(0, width)
+        field = field.to_s.gsub(/\n|\r/, '').slice(0, width) unless field.to_s.include?("\e")
```

Comment:
So, if the field had a \n, we won't remove it if it also has color?  I don't understand that.

Reply:
This will also break the max_width value when it's set really small... To verify, duplicate the test for really long values with max width set, and then add in some color.

In fact, when you do that I expect you are going to get broken field codes, but that can be remedied by adding in a reset `\e[0m;` when the field is truncated.

Reply:
It looks like we use `max_width` in only one spot in `manageiq-release`. With a value of 75.

Someone having ANSI color escape sequences AND a small `max_width` is starting to go towards the probably never gonna happen side of edge cases.

Pulling the to_s and gsub onto a separate line seems simple enough.

Implementing max width and escape sequences requires some pretty heavy lifting. It is a very deep rabbit hole and can we let this bug come out once

Reply:
The gem is generic and supports it as a parameter, so we need to account for it (I use this also for personal projects for example).

Reply:
@Fryguy I'm not sure how to approach this in a way that will be understandable and will not be a bunch of code.

```ruby
rspec spec/core_ext/array/tabelize_spec.rb:37

# given widths = [4,6]
# and we're in the first column

format_field(field: "\033[31mVal\e[0m", width:4)
# => " \033[31mVal\e[0m "
```

|character|pos||
|---|---|---|
|V|5|
|a|6|
|l|7|
|.|12|no 4th character, end of string|

So the method would do a `field.slice(12)` since the 4th character is past the end of the string
Possibly we also want to tack on an extra `"\e[0m"` to be safe.

The only idea I have for this logic is to walk the string skipping escape sequences as we calculate the desired width.

Do you have any better ideas?
Do you think the benefit of this edge case outweighs the code we'll add?